libs/core/langchain_core/indexing/api.py

```diff
@@ -259,6 +256,11 @@ def index(
                        specify a custom vector_field:
                        upsert_kwargs={"vector_field": "embedding"}
             .. versionadded:: 0.3.10
+        scoped_full_cleanup: This argument will be valid only when `claneup` is Full.
```

Comment:
How about we turn this into another clean up mode so we don't introduce another parameter that only works conditionally?

This looks like `cleanup == 'full_scoped'` or something like that.

And we'd document:

1. In the important section -- that full_scoped is a solution to problem w/ batches being a best effort
2. This solution keeps track of the source ids in memory (probably fine for most use cases in terms of memory consumption) -- would require parallelizing for 10M+ docs anyway